temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowLockImpl.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.internal.sync;
+
+import static io.temporal.internal.sync.WorkflowInternal.assertNotReadOnly;
+
+import com.google.common.base.Preconditions;
+import io.temporal.workflow.CancellationScope;
+import io.temporal.workflow.WorkflowLock;
+import java.time.Duration;
+
+class WorkflowLockImpl implements WorkflowLock {
+  private boolean locked = false;
+
+  @Override
+  public void lock() {
+    WorkflowInternal.await(
+        "WorkflowLock.lock",
+        () -> {
+          CancellationScope.throwCanceled();
```

Comment:
Looking at javadoc for `Workflow.await` it should automatically cancel the await and throw if outer scoped is cancelled right?

Reply:
Yes, but this is calling `WorkflowInternal.await` which does not cancel by default. I use `WorkflowInternal.await` to override the wait message.

Reply:
I wonder then if this should just return if cancelled or lock available, and then issue the actual throw outside of the lambda. But so long as the exception propagates to the user, works for me.

Reply:
Yeah it will be propagated, this is how `Workflow.await` is implemented by the java SDK